### PR TITLE
fix tests in edge 17

### PR DESCRIPTION
**Pull request by LavrovArtem**

### CODE REVIEW

File: src/client/sandbox/code-instrumentation/properties/attributes-wrapper.js [hunk 1]
```diff
@@ -2,6 +2,7 @@ import { isHammerheadAttr } from '../../../utils/dom';
 import fnBind from '../../../utils/fn-bind';
 import nativeMethods from '../../native-methods';
 import DomProcessor from '../../../../processing/dom';
+import { attrGetNamedItemInNonEnumerable } from '../../../utils/feature-detection';
```
Comment (miherlosev) [inline src/client/sandbox/code-instrumentation/properties/attributes-wrapper.js]: Do you mean this - `attrGetNamedItemIsNotEnumerable`?

Comment (LavrovArtem) [reply src/client/sandbox/code-instrumentation/properties/attributes-wrapper.js]: yep

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

File: src/client/sandbox/event/focus-blur.js [hunk 2]
```diff
@@ -140,7 +140,7 @@ export default class FocusBlurSandbox extends SandboxBase {
                     this.eventSimulator[bubblesEventType](el, relatedTarget);
                     this.timersSandbox.deferFunction(() => this.eventSimulator[type](el, relatedTarget));
                 }
-                else if (browserUtils.isMSEdge && type === 'blur') {
+                else if (browserUtils.isMSEdge && browserUtils.version < 17 && type === 'blur') {
```
Comment (miherlosev) [inline src/client/sandbox/event/focus-blur.js]: As we will not support the previous MSEdge version (16) and `17` version will update soon on appropriate Windows systems, we can assume that `MSEdge` has only `17` and remove all condition with `browserUtils.version` here and below.

File: src/client/sandbox/event/selection.js [hunk 3]
```diff
@@ -87,8 +88,10 @@ export default class Selection {
                 return selectionSetter();
             }
 
-            // NOTE: setSelectionRange leads to focusing an element only in IE.
-            return selection.wrapSetterSelection(el, selectionSetter, browserUtils.isIE && browserUtils.version < 12);
+            const needFocus = browserUtils.isIE && browserUtils.version < 12 ||
```
Comment (miherlosev) [inline src/client/sandbox/event/selection.js]: Simplify to `const needFocus = browserUtils.isIE11 || browserUtils.isMSEdge`

File: src/client/utils/feature-detection.js [hunk 4]
```diff
@@ -21,3 +21,7 @@ export const hasTouchPoints = browserUtils.isIE && (navigator.maxTouchPoints > 0
 export const isTouchDevice  = (browserUtils.isMobile || browserUtils.isTablet) && hasTouchEvents;
 
 export const hasDataTransfer = !!window.DataTransfer;
+
+// NOTE: In the Edge 17, the getNamedItem method of attributes object in non enumerable
```
Comment (miherlosev) [inline src/client/utils/feature-detection.js]: `... of attributes object is not enumerable`

File: src/client/sandbox/event/selection.js (hunk 3 above)
Comment (miherlosev) [reply src/client/sandbox/event/selection.js]: We decide to leave MSEdge version. But you can simplify `browserUtils.isIE && browserUtils.version < 12` to `browserUtils.isIE11`

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Pull request merged**